subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -206,30 +351,28 @@ static uint8_t notify_handler(struct bt_conn *conn,
 	index = lookup_index_by_handle(handle);
 	if (index < 0) {
 		LOG_DBG("Invalid index");
+
 		return BT_GATT_ITER_STOP;
 	}
 
 	if (length != 0) {
-		err = parse_recv_state(data, length, &recv_state);
+		const uint8_t att_ntf_header_size = 3; /* opcode (1) + handle (2) */
+		uint16_t max_ntf_size;
 
-		if (err != 0) {
-			/* TODO: Likely due to the length.
-			 * Start a read autonomously
-			 */
-			LOG_WRN("Invalid receive state received");
-
-			return BT_GATT_ITER_STOP;
+		if (conn != NULL) {
+			max_ntf_size = bt_gatt_get_mtu(conn) - att_ntf_header_size;
+		} else {
+			max_ntf_size = MIN(BT_L2CAP_RX_MTU, BT_L2CAP_TX_MTU) - att_ntf_header_size;
 		}
 
-		broadcast_assistant.src_ids[index] = recv_state.src_id;
-		broadcast_assistant.past_avail[index] = past_available(conn,
-								       &recv_state.addr,
-								       recv_state.adv_sid);
-
-		if (broadcast_assistant_cbs != NULL &&
-		    broadcast_assistant_cbs->recv_state != NULL) {
-			broadcast_assistant_cbs->recv_state(conn, 0, &
-							    recv_state);
+		broadcast_assistant.long_read_handle = handle;
+		if (length == max_ntf_size) {
+			if (!broadcast_assistant.busy) {
+				net_buf_simple_add_mem(&att_buf, data, length);
+			}
+			long_bap_read(conn, handle);
```

Comment:
Hmm, if we are in the middle of a long read or write (i.e. `broadcast_assistant.busy` is true), then we shouldn't overwrite `broadcast_assistant.long_read_handle = handle;`. 

I guess we may actually need to store a list or array of pending handles to be read.

Reply:
Hmm, do we have the same problem in the bap_unicast_client?

Reply:
No - It has been solved there as we have a timer for each ASE. We may need the same approach here with a timer per receive state